tests/units/inventory_models_test.py

```diff
@@ -355,6 +350,126 @@ def test_inventory_device_invalid(self,test_definition):
                 inventory_device = InventoryDevice( **entity )
             except ValidationError as exc:
                 logging.warning('Error: %s', str(exc))
-                assert True
             else:
                 assert False
+
+    @pytest.mark.parametrize("test_definition", INVENTORY_DEVICE_MODEL, ids=generate_test_ids_dict)
+    def test_inventory_devices_len(self,test_definition):
+        """Test len & append methods for InventoryDevice class.
+
+        Test structure:
+        ---------------
+
+       {
+            "name": "Invalid_Inventory",
+            "input": [
+                {
+                    'host': '1.1.1.1',
+                    'username': 'arista',
+                    'password': 'arista123!',
+                    'established': False,
+                    'url': 'https://demo.io/fake/url'
+                },
+                {
+                    'host': '1.1.1.1',
+                    'username': 'arista',
+                    'password': 'arista123!',
+                    'established': False,
+                    'url': 'https://demo.io/fake/url'
+                }
+            ],
+            "expected_result": "valid"
+        }
+
+        """
+        if test_definition['expected_result'] != 'valid':
+            pytest.skip('Not concerned by the test')
+        inventory_devices = InventoryDevices()
+        for entry in test_definition['input']:
+            inventory_devices.append(InventoryDevice(**entry))
+        if len(test_definition['input']) == len(inventory_devices):
+            logging.info(f'Inventory size is correct ({len(inventory_devices)})')
+        else:
+            logging.error('len function is broken')
+            assert False
```

Comment:
Do we need so many logs in tests?

```suggestion
        assert len(test_definition['input']) == len(inventory_devices):
  ```

Reply:
done